cmake/TPLs/FindTPLMKL.cmake

```diff
@@ -70,10 +70,35 @@
 # pseudorandom number generators.  That's why we require a header
 # file, to access the function declarations.
 
-TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( MKL
-  REQUIRED_HEADERS mkl.h
-  REQUIRED_LIBS_NAMES mkl_rt
-  )
+IF(Kokkos_ENABLE_SYCL)
+  # For OneAPI MKL on GPU, use the CMake target
+  # Temporarily change CMAKE_CXX_COMPILER to icpx to convince MKL to add the DPCPP target
+  # If it sees that CMAKE_CXX_COMPILER is just ".../mpicxx", it won't do this.
+  SET(CMAKE_CXX_COMPILER_PREVIOUS "${CMAKE_CXX_COMPILER}")
+  SET(CMAKE_CXX_COMPILER "icpx")
+  find_package(MKL)
+  SET(CMAKE_CXX_COMPILER "${CMAKE_CXX_COMPILER_PREVIOUS}")
```

Comment:
If you are building with MPI, ``CMAKE_CXX_COMPILER`` is just going to be ``some/path/mpicxx`` (I actually printed this out to debug what MKLConfig.cmake was doing). But AFAIK it's correct to fake it here with ``icpx``, since if we're building with SYCL and enabling MKL, then icpx is the only valid underlying compiler (dpcpp would also be equivalent, but it's deprecated by intel now).